File: backend/api/src/main/java/kpn/api/common/planner/LegBuildParams.java
package kpn.api.common.planner;

import kpn.api.common.planner.LegEnd;
import kpn.core.doc.Storable;

public record LegBuildParams(
  String routeType,
  LegEnd source,
  LegEnd sink,
  Boolean proposed
) implements Storable {
}

/* TODO migrate

  def routeIds: Seq[Long] = {
    legEndRouteIds(source.route) ++ legEndRouteIds(sink.route).distinct
  }

  private def legEndRouteIds(legEndRoute: Option[LegEndRoute]): Seq[Long] = {
    legEndRoute.toSeq.flatMap(_.trackPathKeys).map(_.routeId)
  }

*/

File: backend/api/src/main/scala/kpn/api/common/planner/LegBuildParams.scala
package kpn.api.common.planner

import kpn.api.id.Storable

case class LegBuildParams(
  routeType: String,
  source: LegEnd,
  sink: LegEnd,
  proposed: Boolean
) extends Storable {

  def routeIds: Seq[Long] = {
    legEndRouteIds(source.route) ++ legEndRouteIds(sink.route).distinct
  }

  private def legEndRouteIds(legEndRoute: Option[LegEndRoute]): Seq[Long] = {
    legEndRoute.toSeq.flatMap(_.trackPathKeys).map(_.routeId)
  }
}
